app/src/main/java/com/github/se/travelpouch/ui/authentication/SignIn.kt

```diff
@@ -165,6 +173,40 @@ fun SignInScreen(
                   Text("Sign in with email and password")
                 }
           }
+          if (currentUser != null) {
+            Log.d("SignInScreen", "we're here")
+            LaunchedEffect(Unit) {
+              try {
+                isLoading.value = true
+                methodChosen.value = true
+
+                val gaming =
+                    currentUser.getIdToken(true).await() // to get the result asynchronously
```

Comment:
Main focus is on this line because it is not necessary on Android platform but let's keep it until the app is ready to react to `AuthStateListener`